python/lsst/summit/utils/guiders/seeing.py

```diff
@@ -0,0 +1,517 @@
+
+
+class CorrAnalysis:
+    """
+    Correlation measurement on guider star centroid motions.
+
+    Provides utilities to:
+    - filter detectors by flux threshold,
+    - estimate and correct common drift across detectors, and
+    - measure per-detector variance and pairwise correlations of
+      corrected centroid motions.
+    """
+
+    def __init__(self, stars: pd.DataFrame, expid: int, fluxCut: float = 4e3) -> None:
+        """
+        Initialize correlation analysis from a star catalog.
+
+        Parameters
+        ----------
+        stars : `pandas.DataFrame`
+            Table of star measurements. Must contain columns:
+            ``'detector'``, ``'stamp'`` (time), ``'xfp'``, ``'yfp'``,
+            ``'flux'``, ``'expid'``, and ``'filter'``.
+        expid : `int`
+            Exposure ID to select. Only rows with this expid will be
+            retained.
+        fluxCut : `float`
+            Minimum mean flux required to keep a detector. Default is
+            ``4e3``.
+
+        Raises
+        ------
+        ValueError
+            If no rows with the given ``expid`` are found in ``stars``.
+        ValueError
+            If more than one unique filter is present in the selected
+            exposure.
+        """
+        self.expid: int = expid
+        self.fluxCut: float = fluxCut
+
+        # Early exit if no data
+        mask = stars["expid"].eq(expid)
+        if not mask.any():
+            raise ValueError(f"Stars dataframe has no data with expid={expid}")
+
+        starsFiltered = stars[mask].copy()
+
+        # Wavelength information from LSST bandpass centers (nm).
+        lamMap: dict[str, float] = {
+            "u": 372.4,
+            "g": 480.7,
+            "r": 622.1,
+            "i": 755.9,
+            "z": 868.0,
+            "y": 975.3,
+        }
+        band = starsFiltered["filter"].unique()
+        if len(band) != 1:
+            raise ValueError("cannot have more than one filter per exposure")
+        self.wavelength: float = float(lamMap[str(band[0])])
+
+        # Initialize metrics builder and detrend stars.
+        # These classes/objects are external to this module.
+        dummyNmissingStampPlaceholder = 0
+        metricsBuilder: GuiderMetricsBuilder = GuiderMetricsBuilder(
+            starsFiltered, dummyNmissingStampPlaceholder
+        )
+        self.metrics: pd.DataFrame = metricsBuilder.buildMetrics(self.expid)
+        starsDetrended: pd.DataFrame = metricsBuilder.detrendStars()
+
+        # Build star list per detector with a flux threshold.
+        self.starList: dict[str, pd.DataFrame] = groupStarsPerDetector(starsDetrended, fluxCut)
+        self.detectors: list[str] = list(self.starList.keys())
+
+    def measureVar(self) -> list[float]:
+        """
+        Compute per-detector variances of corrected centroid motions.
+
+        Requires that detrending has already been applied. Variances are
```

Comment:
Is there a way of ensuring that this has been done? What will happen if it hasn't?